project/shacl/modos_schema.shacl.ttl

```diff
@@ -126,291 +140,335 @@ modos:VariantSet a sh:NodeShape ;
             sh:maxCount 1 ;
-            sh:order 7 ;
-            sh:path schema1:description ],
+            sh:nodeKind sh:Literal ;
+            sh:order 6 ;
+            sh:path schema1:name ],
         [ sh:datatype xsd:anyURI ;
             sh:description "The URI from which a resource or dataset was obtained or derived." ;
             sh:maxCount 1 ;
+            sh:nodeKind sh:Literal ;
             sh:order 3 ;
             sh:path modos:source_uri ],
+        [ sh:class modos:ReferenceSequence ;
+            sh:description "Denotes that a sequence belongs to a collection (e.g. a reference genome)." ;
+            sh:nodeKind sh:IRI ;
+            sh:order 1 ;
+            sh:path modos:has_sequence ],
         [ sh:datatype xsd:integer ;
             sh:description "The taxid number describing the taxonomic range of a sample." ;
+            sh:nodeKind sh:Literal ;
             sh:order 2 ;
             sh:path modos:taxon_id ],
         [ sh:datatype xsd:string ;
-            sh:description "The path to access a resource, on a network or local filesystem. Can be a path relative to the root of the digital object, a URL, or an absolute path." ;
-            sh:maxCount 1 ;
-            sh:minCount 1 ;
-            sh:order 0 ;
-            sh:path modos:data_path ],
-        [ sh:datatype xsd:string ;
-            sh:description "A human-readable name for a thing" ;
+            sh:description "A string specifying the release or version of a software or resource." ;
             sh:maxCount 1 ;
-            sh:order 6 ;
-            sh:path schema1:name ],
-        [ sh:description "A unique identifier for a thing" ;
+            sh:nodeKind sh:Literal ;
+            sh:order 4 ;
+            sh:path modos:version ],
+        [ sh:datatype xsd:anyURI ;
+            sh:description "A unique identifier for a thing" ;
             sh:maxCount 1 ;
             sh:minCount 1 ;
+            sh:nodeKind sh:Literal ;
             sh:order 5 ;
             sh:path schema1:identifier ],
         [ sh:datatype xsd:string ;
-            sh:description "A string specifying the release or version of a software or resource." ;
+            sh:description "A human-readable description for a thing" ;
             sh:maxCount 1 ;
-            sh:order 4 ;
-            sh:path modos:version ] ;
+            sh:nodeKind sh:Literal ;
+            sh:order 7 ;
+            sh:path schema1:description ] ;
     sh:targetClass modos:ReferenceGenome .
 
 modos:Sample a sh:NodeShape ;
     sh:closed true ;
     sh:description "A biological sample used in assays. Examples include a whole organism, tissue or cell line." ;
     sh:ignoredProperties ( rdf:type ) ;
-    sh:property [ sh:datatype xsd:string ;
-            sh:description "The organization responsible for collecting a given sample." ;
-            sh:order 4 ;
-            sh:path modos:collector ],
-        [ sh:datatype xsd:integer ;
+    sh:property [ sh:datatype xsd:integer ;
```

Comment:
Though this is maybe a personal opinion of mine: 

if something is an identifier, why require it to be an integer? It's not like you want to do maths with it right? You could even do a regex to check whether the taxid number looks like something you would expect. 

Additionally, if I read your linked issue comment - I'm not sure whether this taxid is now limted to a standard code set.

Reply:
Good point! Actually, I'll make it a URI from https://obofoundry.org/ontology/ncbitaxon.html